packages/core/src/fhirpath/atoms.ts

```diff
@@ -55,13 +55,10 @@ export class SymbolAtom implements Atom {
     if (this.name === '$this') {
       return context;
     }
-    return context
-      .map((e) => this.#evalValue(e))
-      .flat()
-      .filter((e) => e?.value !== undefined) as TypedValue[];
+    return context.flatMap((e) => this.evalValue(e)).filter((e) => e?.value !== undefined) as TypedValue[];
```

Comment:
While researching perf issues, the Node.js profiler revealed that this function is on our CPU hot path.  The reason is that we end up doing about a gajillion `evalFhirPath()` calls when evaluating all search parameters.